daycount_test.go

```diff
@@ -14,558 +15,215 @@ const epsilon = 1.0e-6
+				ActualActual: 326.0/366.0 + 182.0/365.0,
+			},
 		},
 		{
-			"from 2016.03.04 to 2023.06.01",
 			date.New(2016, time.March, 4),
 			date.New(2023, time.June, 1),
-			303.0/366.0 + 6.0 + 151.0/365.0,
+			map[Convention]float64{
+				ActualActual: 303.0/366.0 + 6.0 + 151.0/365.0,
+			},
 		},
 		{
-			"from 2016.03.04 to 2116.03.04",
 			date.New(2016, time.March, 4),
 			date.New(2116, time.March, 4),
-			100.0,
-		},
-	} {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
-
-			assert.InDelta(t, tc.expected, yearFractionActualActual(tc.from, tc.to), epsilon)
-		})
-	}
-}
-
-func Test_yearFractionActualActualAFB(t *testing.T) {
-	t.Parallel()
-
-	for _, tc := range []struct {
-		name     string
-		from     date.Date
-		to       date.Date
-		expected float64
-	}{
-		{
-			"from 2007.12.28 to 2008.02.28",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 28),
-			62.0 / 365.0,
-		},
-		{
-			"from 2007.12.28 to 2008.02.29",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 29),
-			63.0 / 365.0,
-		},
-		{
-			"from 2007.10.31 to 2008.11.30",
-			date.New(2007, time.October, 31),
-			date.New(2008, time.November, 30),
-			1.0 + 30.0/365.0,
-		},
-		{
-			"from 2008.02.01 to 2009.05.31",
-			date.New(2008, time.February, 1),
-			date.New(2009, time.May, 31),
-			1.0 + 120.0/366.0,
-		},
-		{
-			"from 2012.12.28 to 2013.02.28",
-			date.New(2012, time.December, 28),
-			date.New(2013, time.February, 28),
-			62.0 / 365.0,
-		},
-		{
-			"from 2012.02.28 to 2015.01.28",
-			date.New(2012, time.February, 28),
-			date.New(2015, time.January, 28),
-			2.0 + 335.0/366.0,
-		},
-		{
-			"from 1996.01.01 to 1997.01.01",
-			date.New(1996, time.February, 1),
-			date.New(1997, time.January, 1),
-			(366.0 - 31.0) / 366.0,
-		},
-		{
-			"from 2004.02.28 to 2004.03.02",
-			date.New(2004, time.February, 28),
-			date.New(2004, time.March, 2),
-			3.0 / 366.0,
-		},
-	} {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
-
-			assert.InDelta(t, tc.expected, yearFractionActualActualAFB(tc.from, tc.to), epsilon)
-		})
-	}
-}
-
-func Test_yearFractionActualThreeSixty(t *testing.T) {
-	t.Parallel()
-
-	for _, tc := range []struct {
-		name     string
-		from     date.Date
-		to       date.Date
-		expected float64
-	}{
-		{
-			"from 2007.12.28 to 2008.02.28",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 28),
-			62.0 / 360.0,
-		},
-		{
-			"from 2007.12.28 to 2008.02.29",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 29),
-			63.0 / 360.0,
-		},
-		{
-			"from 2007.10.31 to 2008.11.30",
-			date.New(2007, time.October, 31),
-			date.New(2008, time.November, 30),
-			396.0 / 360.0,
-		},
-		{
-			"from 2007.02.01 to 2008.05.31",
-			date.New(2007, time.February, 1),
-			date.New(2008, time.May, 31),
-			485.0 / 360.0,
+			map[Convention]float64{
+				ActualActual: 100.0,
+			},
 		},
 	} {
 		tc := tc
```

Comment:
I think `tc:=tc` is not needed anymore

Reply:
Hmm, not sure about that, I'm still using its fields inside the `t..Run` function - I'm getting linter complaints if I remove it